### PR TITLE
{chem,data}[intel/2020a] h5py v2.10.0, yaff v1.6.0 w/ Python 3.8.2 + add patch for h5py 2.10.0 to avoid triggering MPI_Init at import

### PR DESCRIPTION
**Pull request by boegel**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3106.skitty.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/d7a545aade8d8aa3d0aa7d22e420cc1b for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node2624.swalot.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/b04783a21dd0e0921e2a5776f1a82ab7 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in this PR)
node3416.kirlia.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/7e0cb4260dec2beeeb1f8edb378bc3b9 for a full test report.

File: easybuild/easyconfigs/h/h5py/h5py-2.10.0_avoid-mpi-init.patch [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
++            assert isinstance(info, Info)
++            cdef Py_uintptr_t _comm = _handleof(comm)
++            cdef Py_uintptr_t _info = _handleof(info)
++            H5Pset_fapl_mpio(self.id, <MPI_Comm>_comm, <MPI_Info>_info)
+ 
+ 
+         @with_phil
+@@ -1183,20 +1189,22 @@
+             """
+             cdef MPI_Comm comm
```
Comment (akesandgren) [inline easybuild/easyconfigs/h/h5py/h5py-2.10.0_avoid-mpi-init.patch:57]: I hope that this is also protected with an "IF MPI:"

Comment (boegel) [reply easybuild/easyconfigs/h/h5py/h5py-2.10.0_avoid-mpi-init.patch:61]: It's under the same `IF MPI:` as above.

I didn't come up with this patch myself, it's a backport to h5py 2.10.0 of https://github.com/h5py/h5py/pull/1552

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node2624.swalot.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 2.7.5
See https://gist.github.com/c73f8fa20c78b305e57f11097cbcbefb for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node3416.kirlia.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6240 CPU @ 2.60GHz (cascadelake), Python 2.7.5
See https://gist.github.com/68ae23b2be2257eb4e721c94fb90ccee for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 6 out of 6 (6 easyconfigs in this PR)
node3106.skitty.os - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/72355a32991ebc803fc42d60893da8f2 for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 1 out of 1 (6 easyconfigs in this PR)
vera-c1 - Linux centos linux 7.7.1908, x86_64, Intel Xeon Processor (Skylake), Python 2.7.5
See https://gist.github.com/25a76780fc0ce3141cce88d8d9fa45fc for a full test report.

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 18 out of 18 (6 easyconfigs in this PR)
hebbe-c1 - Linux centos linux 7.7.1908, x86_64, Intel Core Processor (Haswell, no TSX), Python 2.7.5
See https://gist.github.com/88404bcd081a66a6defd5faf32c3c96b for a full test report.

**Comment by Micket** [comment]
Going in, thanks @boegel!

**Pull request merged**